### PR TITLE
Replace futures-signals with eyeball-im in the timeline implementation

### PR DESCRIPTION
**Pull request by jplatte**

### CODE REVIEW

File: bindings/matrix-sdk-ffi/src/api.udl [hunk 1]
```diff
@@ -289,9 +289,7 @@ callback interface TimelineListener {
     void on_update(TimelineDiff update);
 };
 
-interface TimelineDiff {
-    MoveData? move();
-};
```
Comment (Hywan) [inline bindings/matrix-sdk-ffi/src/api.udl:294]: Was is used by ElementX before?

File: crates/matrix-sdk/src/room/timeline/event_handler.rs [hunk 2]
```diff
@@ -808,24 +801,28 @@ pub(crate) fn update_read_marker(
 
             // The read marker can't move backwards.
             if from < to {
-                items_lock.move_from_to(from, to);
+                let item = items.remove(from);
+                // Since the fully-read event's index was shifted to the left
+                // by one position by the remove call above, insert the fully-
+                // read marker at its previous position, rather than that + 1
+                items.insert(to, item);
```
Comment (Hywan) [inline crates/matrix-sdk/src/room/timeline/event_handler.rs:808]: :+1:

File: crates/matrix-sdk/Cargo.toml [hunk 3]
```diff
@@ -76,6 +77,7 @@ futures-core = "0.3.21"
 futures-signals = { version = "0.3.30", default-features = false }
```
Comment (Hywan) [inline crates/matrix-sdk/Cargo.toml:77]: Can we remove it?

```suggestion
```

Edit: Once we have removed it from Sliding Sync too ;-).

File: testing/sliding-sync-integration-test/src/lib.rs [hunk 4]
```diff
@@ -218,9 +218,9 @@ mod tests {
 
             // Test the `Timeline`.
             let timeline = room.timeline().await.unwrap();
-            let mut timeline_stream = timeline.stream();
+            let mut timeline_stream = timeline.stream().await;
 
-            let latest_remote_event = assert_matches!(timeline_stream.next().await, Some(VecDiff::Replace { values }) => {
+            let latest_remote_event = assert_matches!(timeline_stream.next().await, Some(VectorDiff::Append { values }) => {
```
Comment (Hywan) [inline testing/sliding-sync-integration-test/src/lib.rs:223]: :+1:

File: testing/sliding-sync-integration-test/src/lib.rs [hunk 5]
```diff
@@ -291,15 +294,18 @@ mod tests {
...
 
             // The 20th item is a `VecDiff::RemoveAt`, i.e. the first message is removed.
-            assert_matches!(timeline_stream.next().await, Some(VecDiff::RemoveAt { index }) => {
+            assert_matches!(timeline_stream.next().await, Some(VectorDiff::Remove { index }) => {
                 // Index 0 is for day divider. So our first event is at index 1.
                 assert_eq!(index, 1);
             });
 
             // And now, the initial message is pushed at the bottom, so the 21th item is a
             // `VecDiff::Push`.
```
Comment (Hywan) [inline testing/sliding-sync-integration-test/src/lib.rs:303]:
```suggestion
            // `VectorDiff::PushBack`.
```

File: testing/sliding-sync-integration-test/src/lib.rs [hunk 6]
```diff
@@ -291,15 +294,18 @@ mod tests {
             }
 
             // The 20th item is a `VecDiff::RemoveAt`, i.e. the first message is removed.
```
Comment (Hywan) [inline testing/sliding-sync-integration-test/src/lib.rs:296]:
```suggestion
            // The 20th item is a `VectorDiff::Remove`, i.e. the first message is removed.
```

**Review by Hywan — COMMENTED** [review]
It looks really good, bravo!

File: bindings/matrix-sdk-ffi/src/api.udl (hunk 1 above)
Comment (jplatte) [reply bindings/matrix-sdk-ffi/src/api.udl:294]: I'm sure they implemented moving items, otherwise they wouldn't have been able to process all the diffs.

**Pull request merged**